bibtex-completion.el

```diff
@@ -478,6 +493,30 @@ for string replacement."
 (defvar bibtex-completion-cached-notes-keys nil
   "A cache storing notes keys obtained when the bibliography was last parsed.")
 
+(defun bibtex-completion--read ()
+  "Select BibTeX entries in completion system."
+  ;; define a completion function that defaults to completing-read, but can be overridden
+  (list (cdr (assoc (bibtex-completion--completing-read) (bibtex-completion--get-candidates)))))
+
+(defun bibtex-completion--completing-read ()
+  "Read bibtex-completion entries for completion using completing-read."
+  (bibtex-completion-init)
+  (completing-read
+   "BibTeX entries: "
+   (lambda (string pred action)
+     (if (eq action 'metadata)
+         '(metadata
+           (category . bibtex))
+       (complete-with-action action (bibtex-completion--get-candidates) string pred)))))
```

Comment:
There appears to be something subtle wrong here, as I'm not getting the commands in `embark-act`.